int32 dir
int32 duty_cycle
int32 steer
int32 steer_dist